base/applications/cmdutils/more/more.c

```diff
@@ -140,28 +354,118 @@ PagePrompt(PCON_PAGER Pager, DWORD Done, DWORD Total)
     ConClearLine(Pager->Screen->Stream);
 
     /* Ctrl+C or Ctrl+Esc: Control Break */
-    if ((KeyEvent.wVirtualKeyCode == VK_ESCAPE) ||
-        ((KeyEvent.wVirtualKeyCode == L'C') &&
-         (KeyEvent.dwControlKeyState & (LEFT_CTRL_PRESSED | RIGHT_CTRL_PRESSED))))
+    if (fCtrl && ((KeyEvent.wVirtualKeyCode == VK_ESCAPE) ||
+                  (KeyEvent.wVirtualKeyCode == L'C')))
     {
         /* We break, output a newline */
         WCHAR ch = L'\n';
         ConStreamWrite(Pager->Screen->Stream, &ch, 1);
         return FALSE;
     }
 
-    /* 'Q': Quit */
-    // FIXME: Available only when command extensions are enabled.
-    if ((KeyEvent.wVirtualKeyCode == L'Q') &&
-        !(KeyEvent.dwControlKeyState & (LEFT_CTRL_PRESSED | RIGHT_CTRL_PRESSED)))
+    switch (chSubCommand)
```

Comment:
I moved this subCommand block after the Ctrl-C handling.
Also it could even be moved in the block below when extensions are enabled. (Up to you whether you prefer this or not.)

Reply:
OK.